src/main/java/com/example/ludogorieSoft/village/services/VillageService.java

```diff
@@ -73,7 +110,30 @@ public Long createVillageWhitNullValues() {
     }
 
 
-    public VillageDTO updateVillage(Long id, VillageDTO villageDTO) {
+        public VillageDTO updateVillageStatus(Long id, VillageDTO villageDTO) { // approve village status and set admin and date approved
+        Optional<Village> optionalVillage = villageRepository.findById(id);
+        if (optionalVillage.isPresent()) {
+            Village village = optionalVillage.get();
+            village.setName(villageDTO.getName());
+            RegionDTO regionDTO = regionService.findRegionByName(villageDTO.getRegion());
+            village.setRegion(regionService.checkRegion(regionDTO.getId()));
+            village.setPopulationCount(villageDTO.getPopulationCount());
+            village.setPopulation(modelMapper.map(villageDTO.getPopulationDTO(), Population.class));
+
+            AdministratorDTO administratorDTO = authService.getAdministratorInfo();
+
+                village.setAdmin(modelMapper.map(administratorDTO, Administrator.class));
+                village.setStatus(true);
+                village.setDateApproved(now());
+
+            villageRepository.save(village);
+            return modelMapper.map(village, VillageDTO.class);
+        } else {
+            throw new ApiRequestException(ERROR_MESSAGE1 + id + ERROR_MESSAGE2);
+        }
+    }
+
+    public VillageDTO updateVillage(Long id, VillageDTO villageDTO) { //this is used to upload villages file
```

Comment:
Make the method name more descriptive so we can remove the comment. Remember that a more descriptive name is better than adding a comment.

Reply:
just found out what it is used for